dbms/src/IO/ReadHelpers.h

```diff
@@ -623,16 +633,66 @@ inline ReturnType readDateTimeTextImpl(time_t & datetime, ReadBuffer & buf, cons
         return readDateTimeTextFallback<ReturnType>(datetime, buf, date_lut);
 }
 
+template <typename ReturnType>
+inline ReturnType readDateTimeTextImpl(DateTime64 & datetime64, UInt32 scale, ReadBuffer & buf, const DateLUTImpl & date_lut)
+{
+    time_t whole;
+    readDateTimeTextImpl<void>(whole, buf, date_lut);
+    DB::DecimalComponents<DateTime64::NativeType> c{static_cast<DateTime64::NativeType>(whole), 0};
+
+    char separator;
+    if (buf.read(separator))
```

Comment:
Here we always expect separator. But in `writeDateTimeText` we don't write it if `scale = 0`.
So, we can either write separator for scale = 0, or allow to read values without separator.
I think the last case is better, cause we will be able to insert DateTime-formatted data into DateTime64.

You can check next symbol without reading using `*buf.position()` if not `buf.eof()`